src/main/webapp/js/pages/index.js

```diff
@@ -37,18 +37,42 @@ function addScheduleToCalendar(schedule, courseInfo, selected) {
   });
 }
 
-document.querySelector('.course-list').addEventListener('click', () => {
+/**
+ * Returns a list of schedules with the selected courses
+ * @param {Array.<string>} selected JSON Array of the selected classes to
+ *     schedule
+ */
+async function getSchedules(selected) {
+  let response;
+  let scheduleObject;
+  const data = {selectedClasses: selected};
+  try {
+    response = await fetch('/api/scheduler', {
+      method: 'POST',
+      body: JSON.stringify(data),
+    });
+    scheduleObject = await response.json();
+    if (response.ok) {
+      return scheduleObject.schedules;
+    } else {
+      return null;
```

Comment:
I think you need an alert here too. The difference here is that this case handles when the API returns an empty/bad response, whereas the `catch` below handles cases where it didn't even make it to your API.